contents/javascript/Javascript-promises.md

```diff
@@ -90,57 +86,130 @@ These situations correspond to the 3 states of Promises in Javascript.
 Promises provide the ability to specify how the execution of some part of your code would depend on the status of an asynchronous operation. It can now wait for the asynchronous operation to resolve first before doing any work on its result.
 
 ## Quickstart
+Here, we will be showing you some sample code to get started with promises. We understand there is some style of programming you are used to. So here, we are giving you a few examples, first in the style are are familar with, then switching over to promises.
+
+### From callbacks to promises
+Say we have two pretty long and intensive functions, `getData()` and `filterData()`. You will have to get the data from some server using `getData()`, then process it using `filterData()`, all before you can start displaying the results.
+
+And how would such functions be implemented without promises, using the callback method? The callback method ultilizes the fact that we can easily pass functions into javscript as parameters and then use them within the function, effectively "passing" any form of data out, without explicitly returning any value.
 
-Let's return to the setTimeout example at the beginning and rewrite that with promises. 
+We will implement the above use case in the "callback style" (mimicking the long and intensiveness of the functions using `setTimeout`):
 
 ```javascript
+function intenseWait(val, callback) {
+    // return val after a short wait
+    setTimeout(function(){
+        callback(val);
+    }, 1000);
+}
 
-function update() {
-    return new Promise((resolve, reject) => {
-        var x = "I have not been updated";
-        setTimeout(function() {
-            x = "I have been updated";
-            resolve(x);
+/** The very long calls **/
+function getData(callback) {
+    intenseWait('some random data', callback);
+}
+
+function filterData(data, callback) {
+    intenseWait(data.split(' '), callback);
+}
+
+function main() {
+    getData(function(data){
+        filterData(function(filtered){
+            // will print array of splited text
+            console.log(filtered);
+        })
+    });
+}
+
+main();
+```
+
+Now we will rewrite the whole thing in the "promise style". We will be using the same function and variable names, this will let you see how exactly promise works, as opposed to the callback style that you might be more familar with.
```

Comment:
> Now we will rewrite the whole thing in the "promise style".

Let us now rewrite the whole thing using promises. 

>  We will be using the same function and variable names,

Full stop here if not the sentence would be too long to read.

> as opposed to the callback style

as opposed to using callbacks.